components/patient-data/impl/src/test/java/org/phenotips/data/internal/controller/ClinicalStatusControllerTest.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.data.internal.controller;
+
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import net.sf.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.phenotips.data.Patient;
+import org.phenotips.data.PatientData;
+import org.phenotips.data.PatientDataController;
+import org.phenotips.data.SimpleValuePatientData;
+import org.slf4j.Logger;
+import org.xwiki.bridge.DocumentAccessBridge;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.test.mockito.MockitoComponentMockingRule;
+
+import java.util.Collection;
+import java.util.LinkedList;
+
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Test for the {@link ClinicalStatusController} component,
+ * implementation of the {@link org.phenotips.data.PatientDataController} interface
+ */
+public class ClinicalStatusControllerTest
+{
+    @Rule
+    public MockitoComponentMockingRule<PatientDataController> mocker =
+        new MockitoComponentMockingRule<PatientDataController>(ClinicalStatusController.class);
+
+    private Logger logger;
+
+    private DocumentAccessBridge documentAccessBridge;
+
+    private ClinicalStatusController controller;
+
+    @Mock
+    private Patient patient;
+
+    @Mock
+    private XWikiDocument doc;
+
+    @Mock
+    private BaseObject data;
+
+    private final String AFFECTED = "affected";
+
+    private final String UNAFFECTED = "unaffected";
+
+    private final String DATA_NAME = "clinicalStatus";
+
+    @Before
+    public void setUp() throws Exception
+    {
+        MockitoAnnotations.initMocks(this);
+
+        this.controller = (ClinicalStatusController)this.mocker.getComponentUnderTest();
```

Comment:
Why do you need to cast it? You should only work with the interface, which you should always get fresh as `mocker.getComponentUnderTest()`.